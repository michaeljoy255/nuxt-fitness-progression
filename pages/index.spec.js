import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import VueMeta from 'vue-meta'
import PAGE_HEAD from '../constants/page-head'
import IndexPage from './index'

const localVue = createLocalVue()
localVue.use(VueMeta, { keyName: 'head' })

let wrapper

beforeEach(() => {
  const vuetify = new Vuetify()
  wrapper = mount(IndexPage, {
    localVue,
    vuetify,
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Page > index', () => {
  test('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('has correct <head> content', () => {
    expect(wrapper.vm.$metaInfo.title).toBe(PAGE_HEAD.welcome.title)
    const descriptionMeta = wrapper.vm.$metaInfo.meta.find(
      (item) => item.hid === 'description'
    )
    expect(descriptionMeta.content).toBe(PAGE_HEAD.welcome.description)
  })
})
