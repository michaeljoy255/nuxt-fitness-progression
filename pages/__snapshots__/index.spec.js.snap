// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page > index matches the snapshot 1`] = `
"<div class=\\"container\\">
  <h1>Home Page</h1>
</div>"
`;
